src/java/org/apache/cassandra/index/sai/disk/v1/keystore/KeyLookup.java

```diff
@@ -197,43 +197,34 @@ public long clusteredSeekToKey(ByteComparable key, long startingPointId, long en
             updateCurrentBlockIndex(startingPointId);
             resetToCurrentBlock();
 
-            if (compareKeys(currentKey, skipKey) == 0)
-                return startingPointId;
+            // We can return straightaway if the currentPointId is within the requested partition range and the keys match
+            if (currentPointId >= startingPointId && currentPointId < endingPointId && compareKeys(currentKey, skipKey) == 0)
+                return currentPointId;
 
-            if (notInCurrentBlock(startingPointId, skipKey))
-            {
-                long split = (endingPointId - startingPointId) >>> blockShift;
-                long splitPointId = startingPointId;
-                while (split > 0)
-                {
-                    updateCurrentBlockIndex(Math.min((splitPointId >>> blockShift) + split, blockOffsets.length() - 1));
-                    resetToCurrentBlock();
+            long splitStartId = startingPointId;
+            long splitEndId = endingPointId;
 
-                    if (currentPointId >= endingPointId)
-                    {
-                        updateCurrentBlockIndex((endingPointId - 1));
-                        resetToCurrentBlock();
-                    }
-
-                    int cmp = compareKeys(currentKey, skipKey);
-
-                    if (cmp == 0)
-                        return currentPointId;
+            while ((splitEndId - splitStartId) >>> blockShift > 0)
```

Comment:
So this is basically saying the split end is at least the number of points-per-block away, right? (Otherwise we're already at the right block?) Might be worth a quick one-liner comment so casual reader doesn't stumble over the bit shifting, etc.